### DISCUSSION TITLE
Inter-App Broadcast to Multiple Apps

### DISCUSSION THREAD

**Discussion by Jwiggiff** [NightscoutFoundation/xDrip#3662] [Q&A]
I'm currently working on a plugin to [Smartspacer](https://github.com/KieronQuinn/Smartspacer) that displays my BG. Before this, I had the Indentify receiver option set to `info.nightscout.androidaps` since I use AAPS. I first tried deleting this, which resulted in receiving my BG values in my plugin, but not in AAPS. After digging through the code I found [here](https://github.com/NightscoutFoundation/xDrip/blob/2b9f3a821190f36e805e252682975985992849d2/app/src/main/java/com/eveningoutpost/dexdrip/utilitymodels/SendXdripBroadcast.java#L26) that you can use a space-delimited list, so I set the setting to `info.nightscout.androidaps my.package.name` and now I receive bg readings in AAPS but not my plugin. How can I set XDrip to broadcast to multiple apps?

**Comment by Navid200** [comment]
Will you receive readings in your package if you only specify your package without AAPS?
If yes, what happens if you specify your package first?  Will it then only send readings to your package but not to AAPS?

**Comment by Jwiggiff** [comment]
Your answer led me to the problem @Navid200 , thank you! The issue was that, [as mentioned in the docs](https://github.com/KieronQuinn/Smartspacer/wiki/8.-SDK:-Broadcast-Providers#:~:text=Smartspacer%20handles%20registering%20the%20broadcast%20receiver%20for%20you%2C%20and%20will%20send%20your%20plugin%27s%20Broadcast%20Provider%20the%20raw%20intents%20so%20you%20can%20load%20any%20data%20from%20them%2C%20and%20act%20as%20required.), Smartspacer registers the receiver for you, meaning that the package I should've been specifying in xDrip is the Smartspacer package rather than my plugin.
Here was my final setting that worked in both AAPS and my plugin: `info.nightscout.androidaps com.kieronquinn.app.smartspacer`

**Reply by Navid200** [reply]
Great!
